src/session_impl.cpp

```diff
@@ -6593,19 +6594,19 @@ namespace {
 		return m_upnp.get();
 	}
 
-	int session_impl::add_port_mapping(portmap_protocol const t
+	port_mapping_t session_impl::add_port_mapping(portmap_protocol const t
 		, int const external_port
 		, int const local_port)
 	{
-		int ret = 0;
+		port_mapping_t ret{-1};
```

Comment:
do you think this could be a reason for some subtle bug?

Reply:
I suppose so. if both natpmp and upnp are disabled and the user tries to issue add a mapping. I don't think it would have any severe consequences though.